include/odp/api/spec/packet.h

```diff
@@ -2140,6 +2140,32 @@ uint32_t odp_packet_payload_offset(odp_packet_t pkt);
  */
 int odp_packet_payload_offset_set(odp_packet_t pkt, uint32_t offset);
 
+/**
+ * Enable or disable Tx packet drop based on age
+ *
+ * Enable or disable Tx packet drop based on packet age. When enabled, packet will be dropped
+ * if it is in Tx pktout queue or traffic shapers/schedulers for longer than timeout set.
+ *
+ * @param pkt        Packet handle
+ * @param tmo_ns     Timeout to age in nsec
+ *                >0 Aging drop is enabled with specified timeout.
+ *                =0 Aging drop is disabled.
+ *
+ * @see odp_pktio_capability_t::max_tx_aging_tmo_ns
+ */
+void odp_packet_aging_drop_set(odp_packet_t pkt, const uint64_t tmo_ns);
+
+/**
+ * Check if packet has Tx aging drop enabled
+ *
+ * @param pkt Packet handle
+ *
+ * @return Aging drop timeout if enabled.
+ * @retval >0  Aging drop timeout in nano seconds and implies aging drop is enabled.
+ * @retval =0  If Aging drop is disabled.
+ */
+uint64_t odp_packet_has_aging_drop(odp_packet_t pkt);
```

Comment:
"retval =0", should be just "retval 0".

packet_has_xxx calls return flags (0 vs !0). This is not a flag, but a value. So, this should be just:
uint64_t odp_packet_aging_drop(odp_packet_t pkt);

Actually, the metadata value these read/write is not "aging drop", but "aging timeout in nsec". So, I think better name for both functions would be:
odp_packet_aging_tmo()
odp_packet_aging_tmo_set()

Reply:
Ack. Will change it in v5.